### PR TITLE
[Bugfix][Model] Restore causal image SWA for DeepSeek V4.1

### PR DESCRIPTION
**Pull request by Juntian777**
## Summary

Restore causal sliding-window attention for image tokens in DeepSeek V4.1 VL.

V4.1 inherited the image visibility rules used by DeepSeek V4 Vision-Exp: tokens inside an image could attend to earlier image tokens outside the sliding window and to future image tokens. This differs from the [official V4.1 reference implementation](https://huggingface.co/deepseek-ai/DeepSeek-V4.1-Flash/blob/dba1be0a40aa45a94ad051997016db3960a90277/inference/model.py#L410), where `_window_kv` uses `get_window_topk_idxs` and both image and text tokens attend to the same causal window. With a window of 128, query position `p` should see only `[max(0, p - 127), p]` through SWA.

The incorrect widening affects full-image prefill as well as continuation chunks. For example, at the beginning of image span `[1900, 3948)`, the inherited rule exposes future image tokens through position 3947; the V4.1 reference permits only positions 1773 through 1900.

The key line in the [official `get_window_topk_idxs`](https://huggingface.co/deepseek-ai/DeepSeek-V4.1-Flash/blob/dba1be0a40aa45a94ad051997016db3960a90277/inference/model.py#L421) is:

```python
idxs = torch.where(idxs > end, -1, idxs)
```

The reference masks every index greater than the query position with `-1`; this is a causal 128-token window, including the current token. Executing the published `get_window_topk_idxs` at query position 1900 produces exactly 128 indices, from 1773 through 1900, with no future positions.

## Changes

- Disable multimodal prefix-LM/bidirectional-range configuration for V4.1.
- Make the shared SWA metadata builder respect an explicit `is_mm_prefix_lm=False`, keeping V4.1 metadata and paged indices causal.
- Set the V4.1 attention-side image window extension to zero. Image preprocessing, the vision encoder, and `vision_max_n_token` remain unchanged.
- Keep the existing causal gather lengths and workspace sizes. The final diff does not change NVIDIA or AMD workspace allocation.
- Add six regression cases: full prefill and continuation prefill, each with compression ratios `0/1/2`. Verify causal key positions in both gathered and paged indices, gather/CPU-plan consistency, and absence of image visibility extensions.

V4 Vision-Exp retains its existing image visibility behavior.

## Validation

Run on GB200:

```bash
CUDA_VISIBLE_DEVICES=1 .venv/bin/python -m pytest \
  tests/v1/attention/test_deepseek_v4_swa_visible.py \
  tests/kernels/attention/test_flashmla_sparse.py \
  -k 'not test_builder_' -q --tb=short
```

**Result: 30 passed, 3 deselected.** The three existing builder tests require an inaccessible gated Hugging Face configuration. The new regression uses a local configuration and requires no model download.

Changed-file pre-commit checks passed. A local FlashMLA numerical sanity check also matched the causal reference, retaining a gather length of 227 and a first-query SWA length of 128 for `seq_len=4000`, `query_len=100`, and image span `[1900, 3948)`.

Full-model VL accuracy, generation-length evaluation, and AMD runtime validation were not run. The evidence here establishes attention-window correctness at the metadata/kernel level.

## Related work

Open-PR searches for SWA/gather, V4.1/prefill, and V4.1/causal/image found no other matching fix. [#56986](https://github.com/vllm-project/vllm/pull/56986) concerns cross-layer compressed-KV gather reuse; it does not correct the image attention-window semantics addressed here.

## AI assistance

AI assistance was used for analysis, implementation, and validation.

### CODE REVIEW

File: vllm/transformers_utils/configs/deepseek_v41.py [hunk 1]
```diff
@@ -60,11 +60,6 @@ def __init__(
...
-        # prefill + sparse-SWA window widening); the base arch-config
-        # convertor reads this config-level attribute.
-        self.is_mm_prefix_lm = vision_n_layers > 0
-        # The sparse-SWA index kernels widen the window within image spans
-        # in-kernel, so mm-prefix ranges longer than sliding_window must be
-        # kept (they are only consumed by those kernels).
-        self.mm_prefix_clamp_sliding_window = vision_n_layers > 0
+        # Image tokens use causal attention in the language model.
+        self.is_mm_prefix_lm = False
+        self.mm_prefix_clamp_sliding_window = False
```
Comment (Isotr0py) [inline vllm/transformers_utils/configs/deepseek_v41.py]:
```suggestion
```
We can remove them directly.

File: vllm/models/deepseek_v41/attention.py [hunk 2]
```diff
@@ -250,13 +250,8 @@ def __init__(
...
         self.window_size = config.sliding_window
-        # Vision variant: image spans are visible bidirectionally, widening
-        # prefill SWA index rows by up to max_image_tokens columns.
-        self.max_image_tokens = (
-            getattr(config, "vision_max_n_token", 0)
-            if getattr(config, "vision_n_layers", 0) > 0
-            else 0
-        )
+        # V4.1 image tokens use the same causal SWA window as text tokens.
+        self.max_image_tokens = 0
```
Comment (Isotr0py) [inline vllm/models/deepseek_v41/attention.py]: We can further clean up the prefill forwarding to avoid redundant computation now.

**Review by Isotr0py — APPROVED** [review]
